### DISCUSSION TITLE
How to remove a default keybinding?

### DISCUSSION THREAD

**Discussion by mattiaz9** [zed-industries/zed#6552] [Feature Requests]
I have an Italian keyboard and to insert the string interpolation character (\`) I have to type `alt+\` which is currently assigned to copilot suggestion.

I tried to add a custom keybinding for the copilot autosuggestion, but the default one is still active.

So how can I remove the default keybinding?

**Comment by failable** [comment]
https://github.com/zed-industries/community/issues/772

**Comment by JosephTLyons** [comment]
I brought this up with the team today - we'll add a way to remove a binding.  I'll update the #772 when I know more.

**Comment by TomPlanche** [comment]
Hi @JosephTLyons, same problem here.
I have a custom french iso keyboard layout, in order to type `{` i need to do `alt-[` but nothing happens when I do this. 
it is then impossible to code on Zed...

**Reply by SomeoneToIgnore** [reply]
Do things like

```
[
    {
        "context": "Editor",
        "bindings": {
            "alt-v": null,
        }
    }
]
```

work for you?

If yes, we would appreciate a PR into the docs here: https://github.com/zed-industries/zed/blob/main/docs/src/configuring_zed__key_bindings.md

**Reply by TomPlanche** [reply]
I added
```
"alt-[": null,
"alt-]": null,
```
to the `Editor` context.
This totally fixed the problem, thanks !
What do you need in the PR ? A new paragraph to say that we can allow custom keyboard layout using `null` as a value ?

**Reply by SomeoneToIgnore** [reply]
Exactly!
The main part would be to put it so that it is visible for the others who's looking for such feature, and this is where users are much better than developers.

Thank you in advance!

**Reply by TomPlanche** [reply]
The PR is live #7310 !

**Reply by failable** [reply]
@SomeoneToIgnore  The solution does not work for me and I can not remove the `go_to_line::Toggle` command.

```
{
    "context": "Editor && mode == full",
    "bindings": {
      "ctrl-g": null
    }
}
```

**Reply by SomeoneToIgnore** [reply]
Looks like a bug or malformed settings.json

**Reply by failable** [reply]
Some bindings are OK to removed, but some are failed to be removed, e.g. `ctrl-g`. I don't know why.

**Reply by Yevgnen** [reply]
Get the same issue. `ctrl-g` is common key in Emacs stands for "cancel". But I can not unbind it ("go to line") or just rebind it to "cancel".

**Reply by khoaHyh** [reply]
Same here - unable to unbind with `"ctrl-space": null`. Using this keybinding for terminal autosuggest-accept with oh-my-zsh.

**Reply by rakete** [reply]
It's 2025 now and I can actually unbind ctrl-g, but then I can't bind it to anything. I can have it either unbind with `ctrl-g: null` or set to its default `go_to_line::Toggle`

When I try to bind `ctrl-g` to `Editor::Cancel`:
![ctrl-g](https://github.com/user-attachments/assets/f58682d7-a182-425f-ab02-56b21eed60dc)

When I try to unbind `ctrl-g` in `"Editor && mode == full"` first:
![ctrl-g2](https://github.com/user-attachments/assets/d052af48-287f-47a0-ac4f-717e8381d334)

So it is impossible to bind `ctrl-g` to `Editor::Cancel`

**Reply by MrPointer** [reply]
Similar behavior as @rakete showed for me as well, however, for me hitting `ctrl+g` results in the complete **_opposite_** behavior:

<img width="632" alt="SCR-20250604-cptz" src="https://github.com/user-attachments/assets/09c5d6a7-8040-413b-83a8-d8047b92247f" />

I'd expect the behavior to at least be _consistent_, but now I just don't know what to say 😅

**Reply by kdebowski** [reply]
Working on Windows (built from source) I can't unbind "alt-o" which shoud write "ó" in polish. Unfortunately I see a lot of issues with keybindings and support for non-english latin languages. I think this is blocker for many people in terms of using zed as a daily code editor.

**Comment by salzig** [comment]
thanks to this issue I realised I need 
```
  {
    "context": "Editor && mode == full",
    "bindings": {
      "cmd->": null,
      "cmd-<": null
    }
  }
  ```
  to allow window switching on MacOS, so now I'm able to switch between windows by keyboard again. 
  
  Maybe someone else needs this hint. :)

**Comment by lungarella-raffaele** [comment]
I want to disable the minimize shortcut on my sytem. I am on macOs, disabled the global shortcut for minimizing an app using the shortcut "cmd-m", tested on other applications and it's working as intended. Now Zed has it's own shortcut for minimizing the app, I tried disabling it like this, as I understand the context "Workspace" should be the parent of every context. But it does not work. The shortcut I think is correctly disabled because when I open the command palette it does not have the shortcut anymore, it's empty. Am I missing something? I tried adding "cmd-j" and it disabled just fine. 
```
	{
		"context": "Workspace",
		"bindings": {
			"cmd-m": null,
		}
	}
```
